File: python/appwrite/services/files.py
from ..service import Service


class Files(Service):

    def __init__(self, client):
        super(Files, self).__init__(client)

    def upload(self, x, y, z, file):
        """Mock a post request for SDK tests"""

        params = {}
        path = '/mock/tests/files'
        params['x'] = x
        params['y'] = y
        params['z'] = z
        params['file'] = file

        return self.client.call('post', path, {
            'content-type': 'multipart/form-data',
        }, params)

File: node/lib/services/files.js
const Service = require('../service.js');

class Files extends Service {

    /**
     * Mock a post request for SDK tests
     *
     * @param string x
     * @param number y
     * @param array z
     * @param File file
     * @throws Exception
     * @return {}
     */
    async upload(x, y, z, file) {
        let path = '/mock/tests/files';
        
        return await this.client.call('post', path, {
                    'content-type': 'multipart/form-data',
               },
               {
                'x': x,
                'y': y,
                'z': z,
                'file': file
            });
    }
}

module.exports = Files;